### PR TITLE
Add config to release to pypi

### PR DESCRIPTION
**Pull request by johnbradley**
Adds GitHub action to publish the package to pypi when a release is created. 
A PYPI_API_TOKEN secret must be setup in GitHub with a token created in pypi.

Renames the source code directory from `cover/` to `latloncover/` for consistency with the repo and python package name.

### CODE REVIEW

**Comment by johnbradley** [comment]
Waiting for approval for a pypi org for imageomics before proceeding with this PR.

**Comment by johnbradley** [comment]
I read we can transfer an existing pypi package into an organization. Instead of waiting for the org to get approved we can proceed with this PR.

File: setup.py
Comment (egrace479) [inline setup.py:1]: It seems like `setup` should have more information (eg., name of the project, authors, website, etc.) as per the [PyPI example](https://github.com/pypa/sampleproject/blob/db5806e0a3204034c51b1c00dde7d5eb3fa2532e/setup.py).

Comment (johnbradley) [reply setup.py:1]: Python has many different ways to achieve the same thing with package config.
In this case we are using the declarative config option.
See setup.cfg from an earlier commit: 
https://github.com/Imageomics/LatLonCover/blob/2a4d252116eae9abd38f76ba935a34050fe67bdc/setup.cfg#L1-L6
See [setup.cfg documentation](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html) for more

Comment (egrace479) [reply setup.py:1]: Thanks for pointing me to that. Two quick notes:
1. The url should be changed for this repo: https://github.com/Imageomics/LatLonCover.
2. [This line](https://github.com/Imageomics/LatLonCover/blob/2a4d252116eae9abd38f76ba935a34050fe67bdc/setup.cfg#L23C17-L23C17) should have `cover` changed to `latloncover` too, right?

Comment (johnbradley) [reply setup.py:1]: Good catch on the URL. I guess I copied that in and missed it.
The command line tool is `cover` the package and repo name is `LatLonCover`.

Comment (hlapp) [reply setup.py:1]: Shouldn't the package name be more specific too (for the same reason as for which presumably we changed the repo name)?

Comment (johnbradley) [reply setup.py:1]: @hlapp The package name is `latloncover`. Did you mean you want to rename the command line tool to match?

Comment (hlapp) [reply setup.py:1]: Ah yes. Reminder to self not to multitask 😳😉 Though maybe renaming the command line tool for consistency would in fact be a good idea?

**Review by egrace479 — APPROVED** [review]
LGTM. 

Only other note would be a reminder to update the [HF App](https://huggingface.co/spaces/imageomics/LatLonCover) after this goes through so it pulls the new `latloncover` :)

**Pull request merged**
